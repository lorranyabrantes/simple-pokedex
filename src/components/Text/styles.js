import { StyleSheet } from "react-native";
import { COLORS, FONT_FAMILY, FONT_SIZES } from "../../services/styles";

const styles = StyleSheet.create({
  text: {
    color: COLORS.NIGHT_RIDER,
    fontFamily: FONT_FAMILY.roboto.regular,
    fontSize: FONT_SIZES.default,
  },
});

export default styles;
