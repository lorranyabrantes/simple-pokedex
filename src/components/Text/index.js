import React from "react";
import { Text as NativeText } from "react-native";
import styles from "./styles";

const Text = ({ children, style }) => {
  return <NativeText style={[styles.text, style]}>{children}</NativeText>;
};

export default Text;
