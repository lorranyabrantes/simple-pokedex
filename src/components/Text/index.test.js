import React from "react";
import { render } from "@testing-library/react-native";

import Text from "./index";

describe("Text", () => {
  test("should render correctly", async () => {
    const component = render(<Text>Hello</Text>);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test("should apply styles correctly", async () => {
    const component = render(<Text style={{ color: "red" }}>Hello</Text>);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
