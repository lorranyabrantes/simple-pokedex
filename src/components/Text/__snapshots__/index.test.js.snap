// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should apply styles correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontFamily": "Roboto-Regular",
        "fontSize": 14,
      },
      Object {
        "color": "red",
      },
    ]
  }
>
  Hello
</Text>
`;

exports[`Text should render correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
        "fontFamily": "Roboto-Regular",
        "fontSize": 14,
      },
      undefined,
    ]
  }
>
  Hello
</Text>
`;
